evm/executeContract.js

```diff
@@ -0,0 +1,244 @@
+    } else {
+        const ITargetContract = require(targetContractPath);
+        const targetContract = new Contract(targetContractAddress, ITargetContract.abi, wallet);
+        callData = await getCallData(methodName, targetContract, recipient, amount);
+        tokenValue = Number(0);
+    }
+
+    (async () => {
+        try {
+            const result = await contract.executeContract(targetContractAddress, callData, tokenValue);
+            printLog('Function successfully called with return value as: ');
+            printObj(result);
+        } catch (error) {
+            printError('Calling executeContract method failed with Error: ', error);
+        }
+    })();
+}
+
+async function main(options) {
+    const config = loadConfig(options.env);
+    let chains = options.chainNames.split(',').map((str) => str.trim());
+
+    if (options.chainNames === 'all') {
+        chains = Object.keys(config.chains);
+    }
+
+    for (const chain of chains) {
+        if (config.chains[chain.toLowerCase()] === undefined) {
+            throw new Error(`Chain ${chain} is not defined in the info file`);
+        }
+    }
+
+    for (const chain of chains) {
+        const rpc = config.chains[chain.toLowerCase()].rpc;
+        const provider = getDefaultProvider(rpc);
+        const privateKey = options.privateKey;
+
+        if (!isString(privateKey)) {
+            throw new Error('Private Key value is not provided in the info file');
+        }
+
+        const wallet = new Wallet(privateKey, provider);
+        await executeContract(
+            options.pathCallContract,
+            options.pathTargetContract,
+            options.callContractAddress,
+            options.targetAddress,
+            wallet,
+            options.callData,
+            Number(options.nativeValue),
+            options.methodName,
+            options.recipientAddress,
+            Number(options.amount),
+        );
+        saveConfig(config, options.env);
+    }
+}
+
+const program = new Command();
+
+program.name('execute-contract').description('Executes a call to an external contract');
+
+program.addOption(
+    new Option('-e, --env <env>', 'environment')
+        .choices(['local', 'devnet', 'stagenet', 'testnet', 'mainnet'])
+        .default('testnet')
+        .makeOptionMandatory(true)
+        .env('ENV'),
+);
+program.addOption(new Option('-n, --chainNames <chainNames>', 'chain names').makeOptionMandatory(true));
+program.addOption(
+    new Option('-pc, --pathCallContract <pathCallContract>', 'artifact path for the called contract').makeOptionMandatory(true),
+);
+program.addOption(
+    new Option('-pt, --pathTargetContract <pathTargetContract>', 'artifact path for the target contract').makeOptionMandatory(true),
+);
+program.addOption(
+    new Option('-a, --callContractAddress <contractAddress>', 'The contract address in which we will call executeContract function')
+        .makeOptionMandatory(true)
+        .env('CONTRACT_ADDR'),
+);
```

Comment:
I assume the call contract will be one of our deployed contracts, so do we want to pass in the contract name and then get the address from the info file? We could also make this optional and add a default value if we change this to contract name, not address.

Reply:
This generic option was created to support a wider variety of use cases as per the ticket description, if the requirement is specific to default types- yes we should change it.

Reply:
Adding `contractName` and removing `contractAddress`. Added this option earlier to make things more generic, but nice point. :)